src/routes/sessions/[id]/Prompt.svelte

```diff
@@ -1,27 +1,45 @@
 <script lang="ts">
-	import { LoaderCircle, StopCircle, UnfoldVertical } from 'lucide-svelte';
+	import { Brain, LoaderCircle, StopCircle, UnfoldVertical } from 'lucide-svelte';
 	import MessageSquareText from 'lucide-svelte/icons/message-square-text';
 	import Settings_2 from 'lucide-svelte/icons/settings-2';
+	import Trash_2 from 'lucide-svelte/icons/trash-2';
 	import { toast } from 'svelte-sonner';
-	import type { Writable } from 'svelte/store';
+	import { writable, type Writable } from 'svelte/store';
 
 	import LL from '$i18n/i18n-svelte';
 	import Button from '$lib/components/Button.svelte';
 	import ButtonSubmit from '$lib/components/ButtonSubmit.svelte';
 	import Field from '$lib/components/Field.svelte';
 	import FieldSelectModel from '$lib/components/FieldSelectModel.svelte';
 	import FieldTextEditor from '$lib/components/FieldTextEditor.svelte';
-	import { settingsStore } from '$lib/localStorage';
-	import type { Editor } from '$lib/sessions';
+	import { loadKnowledge } from '$lib/knowledge';
+	import { knowledgeStore, settingsStore } from '$lib/localStorage';
+	import type { Editor, Message, Session } from '$lib/sessions';
+	import { generateStorageId } from '$lib/utils';
+
+	import Knowledge from './Knowledge.svelte';
+
+	type KnowledgeAttachment = {
+		fieldId: string;
+		knowledge?: Knowledge;
+	};
 
 	export let editor: Writable<Editor>;
-	export let model: string;
+	export let session: Writable<Session>;
 	export let handleSubmit: () => void;
 	export let stopCompletion: () => void;
 	export let scrollToBottom: (shouldForceScroll: boolean) => void;
 
 	let isOllama = false;
-	$: isOllama = $settingsStore.models?.find((m) => m.name === model)?.api === 'ollama';
+	let attachments: Writable<KnowledgeAttachment[]> = writable([]);
+	let knowledgeOptions: Writable<Knowledge[]> = writable([]);
```

Comment:
I think the type errors we are because we are using a component as the type for the `let knowledgeOptions: Writable<Knowledge[]>` instead of an interface.

This will need to be renamed:

```svelte
import Knowledge from './Knowledge.svelte';
```

Reply:
Oh thank you! That was it. However, for some reason it wasn't enough with renaming the import, so I renamed `Knowledge.svelte` to `KnowledgeSelect.svelte` and now it works fine.